internal/tfdiags/consolidate_warnings.go

```diff
@@ -56,21 +61,30 @@ func (diags Diagnostics) Consolidate(threshold int, level Severity) Diagnostics
 
 		desc := diag.Description()
 		summary := desc.Summary
-		if g, ok := diagnosticGroups[summary]; ok {
+		var consolidationKey string
+		// If the diagnostic has a keyable extra info, use it as the consolidation key,
+		// otherwise use the summary.
+		if key, keyOk := interface{}(diag.ExtraInfo()).(Keyable); keyOk {
```

Comment:
```suggestion
		if key, keyOk := diag.ExtraInfo().(Keyable); keyOk {
```

Reply:
Additionally, it's not enough to have it checked only as `keyOk`. `DeprecationCause` and `VariableDeprecationCause` are `Keyable` by just implementing the interface. But might have an empty key value inside, so we should ensure that the value is not empty.